crystal-map-processor/src/main/java/com/schwarz/crystalprocessor/generation/model/CommonInterfaceGeneration.kt

```diff
@@ -25,8 +27,122 @@ class CommonInterfaceGeneration {
+
+        holder.deprecated?.addDeprecated(typeBuilder)
+
+        if (holder.comment.isNotEmpty()) {
+            typeBuilder.addKdoc(KDocGeneration.generate(holder.comment))
+        }
+
+        for (fieldHolder in holder.allFields) {
+
+            // companionSpec.addProperties(fieldHolder.createFieldConstant())
+            typeBuilder.addProperty(fieldHolder.property(null, holder.abstractParts, false, holder.deprecated))
+        }
+
+        val companionSpec = TypeSpec.companionObjectBuilder()
+        companionSpec.addFunctions(fromMapRepresent(holder))
+        companionSpec.addFunctions(toMapRepresent(holder))
+        companionSpec.addFunctions(TypeConversionMethodsGeneration(useSuspend).generate())
+
+        typeBuilder.addType(companionSpec.build())
+
+        parent.addType(typeBuilder.build())
+    }
+
+    private fun constructorMap(): FunSpec {
+        return FunSpec.constructorBuilder().addModifiers(KModifier.PUBLIC).addParameter("doc", TypeUtil.mutableMapStringAnyNullable()).addStatement("mDoc = ensureTypes(doc).toMutableMap()").build()
+    }
+
+    private fun toMapRepresent(holder: BaseModelHolder): List<FunSpec> {
+        val nullCheck = CodeBlock.builder().beginControlFlow("if(obj == null)").addStatement("return mutableMapOf()").endControlFlow().build()
+        val nullCheckList = CodeBlock.builder().beginControlFlow("if(obj == null)").addStatement("return listOf()").endControlFlow().build()
+
+        return Arrays.asList(
+            FunSpec.builder("toMap").addModifiers(KModifier.PUBLIC)
+                .addParameter("obj", holder.representTypeName.copy(nullable = true)).returns(TypeUtil.mutableMapStringAny())
+                .addAnnotation(JvmStatic::class)
+                .addCode(nullCheck).addStatement("var result = mutableMapOf<%T,%T>()", TypeUtil.string(), TypeUtil.any())
+                .beginControlFlow("obj.mDoc.forEach")
+                .beginControlFlow("if(it.value != null)").addStatement("result[it.key] = it.value!!").endControlFlow()
+                .endControlFlow()
+                .addStatement("return result").build(),
+
+            FunSpec.builder("toMap").addModifiers(KModifier.PUBLIC)
+                .addParameter("obj", TypeUtil.list(holder.representTypeName).copy(nullable = true)).addAnnotation(JvmStatic::class)
+                .returns(TypeUtil.listWithMutableMapStringAny()).addCode(nullCheckList)
+                .addStatement("var result = %T()", TypeUtil.arrayListWithMutableMapStringAny())
+                .addCode(
+                    CodeBlock.builder()
+                        .beginControlFlow("for(entry in obj)")
+                        .addStatement("var temp = mutableMapOf<%T,%T>()", TypeUtil.string(), TypeUtil.any())
+                        .addStatement("temp.putAll(%N.toMap(entry)!!)", GENERATED_REPRESENT_NAME)
+                        .addStatement("result.add(temp)", GENERATED_REPRESENT_NAME).endControlFlow().build()
```

Comment:
Its a minor but I think we don't need the arg "GENERATED_REPRESENT_NAME " here and can leave it out.